src/contracts/atlas/Sorter.sol

```diff
@@ -0,0 +1,199 @@
+//SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.16;
+
+import {IExecutionEnvironment} from "../interfaces/IExecutionEnvironment.sol";
+import {IAtlas} from "../interfaces/IAtlas.sol";
+import {IEscrow} from "../interfaces/IEscrow.sol";
+import {IProtocolControl} from "../interfaces/IProtocolControl.sol";
+
+import "../types/CallTypes.sol";
+
+import {CallBits} from "../libraries/CallBits.sol";
+import {CallVerification} from "../libraries/CallVerification.sol";
+
+contract Sorter {
+
+    address immutable public atlas;
+    address immutable public escrow;
+
+    constructor(address _atlas, address _escrow) {
+        atlas = _atlas;
+        escrow = _escrow;
+    }
+
+    struct SortingData {
+        uint256 amount;
+        bool valid;
+    }
+
+    function sortBids(
+        UserCall calldata userCall, 
+        SearcherCall[] calldata searcherCalls
+    ) external view returns (SearcherCall[] memory) {
+
+        ProtocolCall memory protocolCall = IProtocolControl(userCall.metaTx.control).getProtocolCall();
+
+        uint256 count = searcherCalls.length;
+
+        (SortingData[] memory sortingData, uint256 invalid) = _getSortingData(
```

Comment:
At first glance got confused why invalid was a uint and not a bool, so maybe changing name to invalidCount is better for readability? 
```suggestion
        (SortingData[] memory sortingData, uint256 invalidCount) = _getSortingData(
```

Reply:
great point - let's do it.